42-cellular-smoke-detector-retrofit/README.md

```diff
@@ -0,0 +1,184 @@
+
+<p align="center">
+<img src="images/Optoisolator_Pinout.svg.jpg"/>
+</p>
+
+Optocouplers are useful for passing signals between circuits operating at different voltages. In this case, we need pass the 9V signal from the smoke detector interconnect onto the Notecard without damaging the Notecard. To do so, we'll use the optocoupler as a kind of switch. When the interconnect wire goes high to 9V, the optocoupler will effectively close a switch that connects one of the Notecard's GPIO pins to ground. When the interconnect wire is low at 0V, the switch will be open, and an internal pull-up on the Notecard's GPIO will keep the logic level high.
+
+#### Choosing a Resistor Value
+
+If you hook up the optocoupler to the interconnect wire with no series resistor, you risk frying the chip. You must limit the forward current, the current flowing through the optocoupler's LED, with a resistor in series with the optocoupler's LED. However, if your resistance is too high, the forward current will be insufficient to get the output transistor to saturate. Looking at [the PC817 datasheet](https://www.farnell.com/datasheets/73758.pdf), we can see that the maximum forward current under "Absolute Maximum Ratings" is 50mA. In the "Electro-optical Characteristics" table, we see a forward current of 20mA used as a condition for several of the values in the table. As such, designing the circuit for a 20mA forward current is a good place to start.
+
+To estimate the resistance needed to achieve a particular forward current, use Ohm's Law: V = IR. Rearrange it to get R = V/I. V is the voltage drop over your resistor. Looking at the datasheet, we see that the forward voltage, which is the voltage dropped over the LED, is typically 1.2V if the forward current is 20mA. With 6.7V coming from the source, 6.7V - 1.2V = 5.5V will be dropped over our resistor. Plugging this voltage and 20mA into R = V/I, we get R = 5.5V / 20e-3 = 275 Ω.
+
+From our testing of this particular PC817 from Gikfun ([linked above](#you-will-need)), we were able to saturate the transistor with just 0.8mA of forward current. To achieve this with a source voltage of 6.7V, we used 2 resistors in series: 1kΩ and 4.7kΩ. If you plug 5.7kΩ into the calculations above, you won't get exactly 0.8mA of forward current. In practice, however, this is what we observed when using a digital multimeter to measure the current.
+
+### Breadboarding
+
+Setting aside the Notecard and Notecarrier, which you'll assemble and configure shortly, build the circuit on the breadboard:
+
+<p align="center">
+  <img src="images/breadboard_no_notecarrier.jpg"/>
+</p>
+
```

Comment:
Would be nice with a Fritzing diagram too.

Reply:
Yeah, that would be useful. I'll add one.